src/python/pants/python/python_setup.py

```diff
@@ -122,7 +122,9 @@ def register_options(cls, register):
             "strings are supported: "
             '"<PATH>" (the contents of the PATH env var), '
             '"<PEXRC>" (paths in the PEX_PYTHON_PATH variable in a pexrc file), '
-            '"<PYENV>" (all python versions under $(pyenv root)/versions).',
+            '"<PYENV>" (all python versions under $(pyenv root)/versions).'
+            'Environment variables will be expanded, '
+            'e.g. the value "$\{HOME\}/example" may be used.',
```

Comment:
As mentioned in the description: I think that the help on this option should specifically mention which variables we guarantee to expand. Maybe just HOME, or whatever we need for this usecase.

Reply:
I think @silverguo wanted to be able to use `PYENV_ROOT` so that he can say `PYENV_ROOT/versions/3.7.5/bin`. Only allowing `HOME` wouldn't help because `PYENV_ROOT` can be any arbitrary location, like in our own Dockerfiles we set it up at `/pyenv-docker`.

Reply:
Sure. As I said: we just need to be explicit about which variables we support. If that's the one we need to support, let's do it.

Reply:
@silverguo would it solve your use case if you changed this PR to instead expose the special symbol `<PYENV_ROOT>`, rather than allowing for any arbitrary environment variable? We would end up exposing `<PATH>`, `<PYENV>`, `<PYENV_ROOT>`, and `<PEXRC>`. 

The reason this would be good is that we try to keep things more restrictive by default, generally. It's much easier to add flexibility down the road when we find we need it, rather than to take away flexibility when we realize it's a problem.

Reply:
Hi @Eric-Arellano @stuhood , making it more restrictive makes lots of sense.

I think it should be safe to have a new special symbol `<PYENV_LOCAL>`, which is supposed to only use the python specified in `BUILD_ROOT/.python-version` file.

I made some changes to `python_setup.py`, but I don't know what's the best practice to add a test for it, since I need to fake a `.python-version` file under BUILD_ROOT.

If the idea of this new special symbol works for you, could you tell me how to add a test around it ?